extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -206,6 +210,13 @@ export class AmpAd3PImpl extends AMP.BaseElement {
    */
   onLayoutMeasure() {
     this.isInFixedContainer_ = !isAdPositionAllowed(this.element, this.win);
+    /** detect ad containers, add the list to element as a new attribute */
+    if (!this.container_) {
+      this.container_ = getAdContainer(this.element);
```

Comment:
We can use `undefined`  and `null` to differentiate "container not initialized" vs "no container"